internal/clients/http/azure/azure_service_client.go

```diff
@@ -0,0 +1,147 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+)
+
+type serviceClient struct {
+	credential *azidentity.ClientSecretCredential
+}
+
+func init() {
+	clients.GetServiceAzureClient = newServiceClient
+}
+
+func newServiceClient(ctx context.Context) (clients.AzureService, error) {
+	opts := azidentity.ClientSecretCredentialOptions{}
+	identityClient, err := azidentity.NewClientSecretCredential(config.Azure.TenantID, config.Azure.ClientID, config.Azure.ClientSecret, &opts)
+	if err != nil {
+		return nil, fmt.Errorf("unable to init Azure credentials: %w", err)
+	}
+
+	return &serviceClient{
+		credential: identityClient,
+	}, nil
+}
+
+func (c *serviceClient) RegisterInstanceTypes(ctx context.Context, instanceTypes *clients.RegisteredInstanceTypes, regionalTypes *clients.RegionalTypeAvailability) error {
```

Comment:
This function is 90 lines long. I suggest splitting this, there are some logic parts that can be separated. Especially the huge for loop. Moreover, on the line 118, the for loop is nested on level four.
We could have a helper function to return `instanceType.Cores` value. *c.name comparisons seem like a good example for decomposition as well.

Reply:
That is a fair point, I extracted a new function called `typeFromSKU` that moves probably half of the function into a separate one. That cut's it at half, a good starting point I guess.

Unfortunately, I accidentally merged it into the same commit, this should have been done in a separate commit, or separate PR ideally. Sorry for that, it is gonna be hard to follow the change now since the code moved from a different file, but it was essentially just extraction using GoLand "extract method" feature plus I renamed variable `v` to `resourceSKU` which is much more appropriate.